### PR TITLE
Sharding support

### PR DESCRIPTION
**Pull request by ramidzkh**
Adds sharding support.

This system has many drawbacks, as all modules are created for every single shard. This also brings up the question of when will the configuration system be added, as the amount of shards depends on the environment variables.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "no-console": "off",
     "class-methods-use-this": "off",
     "require-await": "error",
+    "no-plusplus": "off",
```
Comment (ramidzkh) [inline .eslintrc.json]: This is required for the `++` operators. A work around would be `+= 1`

File: src/index.js [hunk 2]
```diff
@@ -3,12 +3,32 @@ const path = require('path');
 const { Client } = require('discord.js');
 const { Handler } = require('./handler');
 
-const client = new Client({ disableEveryone: true });
-const handler = new Handler(client);
+// TODO: Implement a proper config
+const config = {
+  token: process.env.TOKEN,
+  shards: parseInt(process.env.SHARDS, 10) || 1,
```
Comment (ramidzkh) [inline src/index.js]: Assumes base 10, as ESLint forces a radix parameter.

File: src/index.js [hunk 3]
```diff
@@ -3,12 +3,32 @@ const path = require('path');
 const { Client } = require('discord.js');
 const { Handler } = require('./handler');
 
-const client = new Client({ disableEveryone: true });
-const handler = new Handler(client);
+// TODO: Implement a proper config
+const config = {
```
Comment (ramidzkh) [inline src/index.js]: Don't do this in the first place?

File: .eslintrc.json (hunk 1 above)
Comment (SebastiaanYN) [inline .eslintrc.json]: The reason for the `no-plusplus` rule is explained here: [Variables - unary increment decrement](https://github.com/airbnb/javascript#variables--unary-increment-decrement). If you feel like this rule should be disabled open a different PR for it and provide a reason.

File: src/index.js [hunk 4]
```diff
@@ -3,12 +3,32 @@ const path = require('path');
...
+  token: process.env.TOKEN,
+  shards: parseInt(process.env.SHARDS, 10) || 1,
+};
 
-handler.load(path.join(__dirname, './modules'), {
-  client,
-  commandHandler: handler,
-});
+for (let shard = 0; shard < config.shards; shard++) {
```
Comment (SebastiaanYN) [inline src/index.js]: This could just be changed to `shard += 1`.

File: src/index.js (hunk 2 above)
Comment (SebastiaanYN) [reply src/index.js]: Explained [here](https://github.com/airbnb/javascript#coercion--numbers) it's also possible to do `Number(process.env.SHARDS)`.

**Comment by ramidzkh** [comment]
I'll do this later
